server/src/com/thoughtworks/go/server/service/materials/GitPoller.java

```diff
@@ -23,22 +23,29 @@
 import com.thoughtworks.go.config.materials.git.GitMaterial;
 import com.thoughtworks.go.domain.materials.Modification;
 import com.thoughtworks.go.domain.materials.Revision;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.jruby.RubyProcess;
 
 public class GitPoller implements MaterialPoller<GitMaterial> {
 
     @Override
     public List<Modification> latestModification(GitMaterial material, File baseDir, SubprocessExecutionContext execCtx) {
+        material = material.withShallowClone(isServerSideShallowCloneOn());
         return material.latestModification(baseDir, execCtx);
     }
-
     @Override
     public List<Modification> modificationsSince(GitMaterial material, File baseDir, Revision revision, SubprocessExecutionContext execCtx) {
-        return material.modificationsSince(baseDir, revision, execCtx);
+        material = material.withShallowClone(isServerSideShallowCloneOn());
+        return material.withShallowClone(true).modificationsSince(baseDir, revision, execCtx);
```

Comment:
The flag is being set to true always, the previous statement does the right thing - Did you leave this by mistake?